docs/deep-dives/mobile/enrollment-state-machine.md

```diff
@@ -0,0 +1,176 @@
+
+#### Opt Out
+
+The client has opted out of experimentation.
+
+#### EnrollmentsPaused
+
+The recipe has paused enrollments.
+
+#### Feature Conflict
+
+The feature(s) the recipe targets a) are being experimented on by another recipe of the same type in which the client is already enrolled<sup><a href="#notes">[3]</a></sup>, and b) are not enabled for coenrollment.
+
+### Disqualified
+
+:::info
+This state and its reasons are for enrollments that were previously in one of the `Enrolled` states, but are no longer enrolled.
+:::
+
+#### Not Targeted
+
+_See [NotEnrolled::NotTargeted](#not-targeted)_
+
+#### Not Selected _(NEW 7/7/2023)_
+
+_See [NotEnrolled::NotSelected](#not-selected)_
+
+#### Opt Out
+
+The client manually opted out of the recipe<sup><a href="#notes">[2]</a></sup>.
+
+#### Error
+
+The client threw an error during enrollment re-evaluation.
+
+### Was Enrolled
+
+The recipe has ended, and the client was enrolled at some point.
+The `Enrolled` and `Disqualified` statuses end up here when a recipe ends.
+
+### Error
+
+The client threw an error during enrollment evaluation<sup><a href="#notes">[4]</a></sup>.
+
+### Notes
+
+1. The explicit order in which recipe criteria are evaluated is as follows:
+    1. Feature conflicts
+    2. Targeting
+    3. Bucketing
+2. This state can be reached via Firefox mobile's secret menu.
+3. Experiments and rollouts do not share feature conflicts.
+   A client can be enrolled in up to one rollout and one experiment for a given feature, unless that feature has been enabled for coenrollment.
+4. This status will only contain enrollments for recipes in which the client has never been enrolled.
+   If the client is enrolled but throws an error during enrollment re-evaluation it will be moved to [Disqualified::Error](#error)
+
+## State Diagrams
+
+The following state diagrams describe the interactions between states for the enrollment state machine in the mobile Nimbus SDK.
+
+### Experiments
+```mermaid
+stateDiagram-v2
+    direction LR
+
+    [*] --> enrolled
+    [*] --> not_enrolled
+
+    enrolled: Enrolled
+    not_enrolled: Not Enrolled
+    disqualified: Disqualified
+    was_enrolled: Was Enrolled
+    error: Error
+
+    enrolled --> disqualified
+
+    state on_experiment_ended <<join>>
+    enrolled --> on_experiment_ended
+    disqualified --> on_experiment_ended
+    on_experiment_ended --> was_enrolled: on experiment ended
+
+    was_enrolled --> [*]: on garbage\ncollect (1yr)
+
+    state if_enrollment_not_paused <<choice>>
+    not_enrolled --> if_enrollment_not_paused
+    error --> if_enrollment_not_paused
+    if_enrollment_not_paused --> enrolled: if enrollment is not paused
+
+    state on_error <<join>>
+    not_enrolled --> on_error
+    enrolled --> on_error
+    on_error --> error: on error
+```
+
+### Rollouts
+
+Rollouts and experiments have _almost_ the same state machine, but rollouts have additional functionality surrounding disqualified states.
```

Comment:
At this level, I don't _think_ there is a difference yet between rollouts and experiments.

Part of this exercise was to establish what is in the code, rather than what is _intended_ to be in the code, then update this diagram after you've made the changes.

Reply:
Ah sorry I misunderstood that! Yep I'll pull the rollout-specific stuff out and make sure it reflects the current state.

Reply:
I think even once https://github.com/mozilla/application-services/pull/5716 lands, the update then is to added a single edge from Disqualified --> Enrolled, with the label to the effect of "rollouts only".